Midterm markdown refactor (#190)
yagpdb-cc/yagpdb-cc: suggestion/README.md

# Suggestion System
A feature-rich suggestion system, condensed into a single custom command.<br>
This is similar to the current system used in the YAGPDB Support Server.

## Features
* Logging channels
* Management commands
* Cooldown to prevent spam

## Installing 
In case you need a refresher, or don't know how to add a custom command, please read [this](https://learn.yagpdb.xyz/the-custom-command-interface) tutorial.

For now, we will just add the system to your server and elaborate later on the various configuration variables as well as commands and subcommands.

Copy the entire script from [suggestion.go.tmpl](suggestion.go.tmpl) and add it as a new custom command. Set the trigger type to RegEx and paste the following as the trigger:

###### Suggestion System Trigger
```
(?i)\A(\-\s?|<@!?204255221017214977>\s*)((del(ete)?|edit)?suggest(ion)?|(sa|suggestadmin)\s+((?:mark)?dupe|deny|implement(ed)?|archive|approved?|comment))(\s+|\z)
```

| ⚠ Make sure to replace the `-` at the beginning of the trigger with your server's prefix.
| ---- |

> You will not be able to save yet, we have to do one minor tweak: Remove the large leading comment as well as the second comment:

Reviewer: ```suggestion
You will not be able to save yet, as the custom command code is slightly over the limit as is. To fix this, we have to do one minor tweak: remove the large leading comment as well as the second comment:
```

Brief explanation as to why it does not save.

-
-	Recommended trigger: Regex trigger with non case sensitive and trigger 
-  `\A(\-\s?|<@!?204255221017214977>\s*)((del(ete)?|edit)?suggest(ion)?|(sa|suggestadmin)\s+((?:mark)?dupe|deny|implement(ed)?|archive|approved?|comment))(\s+|\z)`
-   Note: If your prefix is not `-` replace the `-` at the start of the trigger with your prefix.
-	P.S. - REMOVE This comment so that you can save the command. Otherwise youll get an error!
- */}}
- 
- {{/* CONFIGURATION AREA STARTS */}}
{{$Suggestion_Channel:=737324313341853796}}
```
Remove the text you see here as red from your custom command. Now you can save, which you should definitely do.

## Configuration
Okay, there are still a few things to do before your suggestion system is set up, which are the configuration variables. Please take your time to read through their description before making any changes.

### Channels 
| ℹ Among these variables, all channels can be separate or the same. |
| ---- |

* `$Suggestion_Channel`<br>
   This is the main channel, where all suggestions made through the command will show up - This is the `#suggestions` channel on the support server.
* `$Logging_Channel`<br>

>    This is the channel where the authors are being notified if a suggestion was denied, approved, implemented, or marked as dupe. In the YAGPDB server, this is the `#suggestion-discussion` channel.

Reviewer: ```suggestion
   This is the channel where the authors are notified if a suggestion was denied, approved, implemented, or marked as dupe. In the YAGPDB server, this is the `#suggestion-discussion` channel.
```

* `$Implemented_Channel`<br>
   The channel where suggestions which have been marked as implemented are sent. This provides a good way to organise all implemented suggestions into a separate channel. On the support server, you can find this channel as `#implemented-suggestions`.
* `$Approved_Channel`

>    Where approved suggestions are being sent to. This is not quite equal to implemented suggestions: You can see approved suggestions as "being worked on", and implemented suggestions as "this is now a feature". In the Support Server, we log those under `#implemented-suggestions` as well.

Reviewer: ```suggestion
   Where approved suggestions are being sent to. This is not the same as implemented suggestions: you can see approved suggestions as "being worked on", and implemented suggestions as "this is now a feature". In the Support Server, we log those under `#implemented-suggestions` as well.
```
Reads slightly better.

| ---- |

## Commands
| ℹ Precede all commands covered in the following sections either with your prefix or mention YAGPDB. |
| ---- |

This section documents the commands and subcommands of this system, alongs their use-case and usage.

### For everyone
* `suggest`<br>
   Syntax: `suggest <Suggestion:Text>`<br>
   Use: Used to submit a new suggestion.
* `deletesuggestion`<br>
   Syntax: `deletesuggestion <Suggestion-ID:Whole Number>`<br>
   Use: Delete the suggestion with the given ID. Can be used by mods to force-delete a suggestion.
*  `editsuggestion`<br>
   Syntax: `editsuggestion <Suggestion-ID:Whole Number> <Edited Suggestion:Text>`<br>
   Use: Edit the suggestion with the given ID. Replaces the old text entirely with the new text.

### For Mods / Admins
| ℹ All these commands are to be preceded with `sa` or `suggestadmin`. |
| ---- |

* `deny`<br>
   Syntax: `deny <Suggestion-ID:Whole Number> [Reason:Text]`<br>
   Use: Deny a suggestion and notify the author that their suggestion has been deleted, along with the optional reason for denial.
* `dupe`<br>
   Syntax: `dupe <Suggestion-ID:Whole Number> <Old Suggestion-ID:Whole Number>`<br>

>    Use: Mark the first given suggestion as dupe of the second suggestion, delete the duplicate and inform the author of the duplicate.

Reviewer: ```suggestion
   Use: Mark the first given suggestion as a dupe of the second suggestion, delete the duplicate and inform the author of the duplicate.
```

* `approve`<br>
   Syntax: `approve <Suggestion-ID:Whole Number> [Comment:Text]`<br>

>    Use: Approve of a suggestion and log it to the channel for approved suggestions and notify the author. Adds a record of who approved the suggestion.

Reviewer: ```suggestion
   Use: Approve a suggestion and log it to the channel for approved suggestions and notify the author. Adds a record of who approved the suggestion.
```
